langchain4j-mcp/src/main/java/dev/langchain4j/mcp/McpToolProvider.java

```diff
@@ -68,7 +75,14 @@ public McpToolProvider.Builder mcpClients(List<McpClient> mcpClients) {
          * The list of MCP clients to use for retrieving tools.
          */
         public McpToolProvider.Builder mcpClients(McpClient... mcpClients) {
-            this.mcpClients = Arrays.asList(mcpClients);
+            return mcpClients(Arrays.asList(mcpClients));
+        }
+
+        /**
+         * The predicate to filter MCP provided tools.
+         */
+        public McpToolProvider.Builder mcpToolsFilter(BiPredicate<McpClient, ToolSpecification> mcpToolsFilter) {
```

Comment:
```suggestion
        public McpToolProvider.Builder filter(BiPredicate<McpClient, ToolSpecification> filter) {
```
nit: we are in the `McpToolProvider` class, so both "mcp" and "tool" parsts seem to be repetitive

Reply:
+1, fixed.